lms/static/js/spec/groups/views/cohorts_spec.js

```diff
@@ -0,0 +1,657 @@
+                    expect($(this).text().trim()).toBe(expectedDetails[index]);
+                });
+            };
+
+            verifyHeader = function(expectedCohortId, expectedTitle, expectedCount) {
+                var header = cohortsView.$('.cohort-management-group-header');
+                expect(cohortsView.$('.cohort-select').val()).toBe(expectedCohortId.toString());
+                expect(cohortsView.$('.cohort-select option:selected').text()).toBe(
+                    interpolate_text(
+                        '{title} ({count})', {title: expectedTitle, count: expectedCount}
+                    )
+                );
+                expect(header.find('.title-value').text()).toBe(expectedTitle);
+                expect(header.find('.group-count').text()).toBe(
+                    interpolate_ntext(
+                        '(contains {count} student)',
+                        '(contains {count} students)',
+                        expectedCount,
+                        {count: expectedCount}
+                    )
+                );
+            };
+
+            unknownUserMessage = function (name) {
+                return "Unknown user: " +  name;
+            };
+
+            beforeEach(function () {
+                setFixtures('<ul class="instructor-nav"><li class="nav-item"><<a href data-section="membership" class="active-section">Membership</a></li></ul><div></div>');
+                TemplateHelpers.installTemplate('templates/instructor/instructor_dashboard_2/cohorts');
+                TemplateHelpers.installTemplate('templates/instructor/instructor_dashboard_2/cohort-form');
+                TemplateHelpers.installTemplate('templates/instructor/instructor_dashboard_2/cohort-selector');
+                TemplateHelpers.installTemplate('templates/instructor/instructor_dashboard_2/cohort-editor');
+                TemplateHelpers.installTemplate('templates/instructor/instructor_dashboard_2/notification');
+                TemplateHelpers.installTemplate('templates/file-upload');
+            });
+
+            it("Show an error if no cohorts are defined", function() {
+                createCohortsView(this, {cohorts: []});
+                verifyMessage(
+                    'You currently have no cohort groups configured',
+                    'warning',
+                    'Add Cohort Group'
+                );
+
+                // If no cohorts have been created, can't upload a CSV file.
+                expect(cohortsView.$('.wrapper-cohort-supplemental')).toHaveClass('is-hidden');
+            });
+
+            it("Syncs data when membership tab is clicked", function() {
+                createCohortsView(this, {selectCohort: 1});
+                verifyHeader(1, 'Cat Lovers', catLoversInitialCount);
+                $(cohortsView.getSectionCss("membership")).click();
+                AjaxHelpers.expectRequest(requests, 'GET', '/mock_service/cohorts');
+                respondToRefresh(1001, 2);
```

Comment:
I found this a little hard to read. Maybe rename it to to more clearly reflect that it's returning cohort json, and that the arguments indicate cat count and dog count respectively? Maybe something like `respondWithCohortJson({catCount: 1001, dogCount: 2})`